### PR TITLE
[Extensions] Allow default log4j behavior for users of SDK

### PR DESCRIPTION
**Pull request by vibrantvarun**
### Description
Fix to bug
[824](https://github.com/opensearch-project/opensearch-sdk-java/issues/824#issuecomment-1636226463)

Steps to fix
[https://github.com/opensearch-project/opensearch-sdk-java/issues/824#issuecomment-1636226463](https://github.com/opensearch-project/opensearch-sdk-java/issues/824#issuecomment-1636226463)

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -146,7 +146,7 @@ dependencies {
     def javaxVersion  = "1"
     def guavaFailureAccessVersion = "1.0.1"
     def aopallianceVersion = "1.0"
-    def slf4jVersion = "1.7.36"
+    //def slf4jVersion = "1.7.36"
```
Comment (dbwiddis) [inline build.gradle:149]: Looks great but just remove instead of commenting :)

File: build.gradle [hunk 2]
```diff
@@ -199,8 +199,10 @@ dependencies {
             force("org.apache.logging.log4j:log4j-api:${log4jVersion}")
             force("org.apache.logging.log4j:log4j-core:${log4jVersion}")
             force("org.apache.logging.log4j:log4j-jul:${log4jVersion}")
-            force("org.slf4j:slf4j-api:${slf4jVersion}")
+            //force("org.slf4j:slf4j-api:${slf4jVersion}")
```
Comment (dbwiddis) [inline build.gradle:202]: ditto

File: src/main/resources/log4j2-sdk.xml [hunk 3]
```diff
@@ -13,6 +13,5 @@
             <AppenderRef ref="ConsoleAppender" />
             <AppenderRef ref="FileAppender"/>
         </Root>i
-    <logger name="io.netty.util" level="ERROR"/>
```
Comment (dbwiddis) [inline src/main/resources/log4j2-sdk.xml:16]: Can you also remove the stray `i` in the line above this?

**Review by dbwiddis — COMMENTED** [review]
Looks good!   Can we fully delete instead of just commenting?

File: build.gradle [hunk 4]
```diff
@@ -199,8 +199,10 @@ dependencies {
...
         }
+
+        exclude group: 'org.slf4j'
```
Comment (owaiskazi19) [inline build.gradle:205]: Is this a transitive dependency coming from some other dependency?

**Comment by owaiskazi19** [comment]
The change is failing the integ test though:

```
 org.opensearch.sdk.sample.helloworld.rest.TestHelloWorldIT > testInitializeExtension FAILED
    java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory

File: build.gradle (hunk 4 above)
Comment (vibrantvarun) [reply build.gradle]: Comment addressed and verified by dan

Comment (dbwiddis) [reply build.gradle]: @owaiskazi19 this turned out to be a gnarlier issue than I thought.  I'll update the issue and create a new one.

TLDR we are keeping both imports as is.  We are just moving the log4j xml file.

**Review by dbwiddis — APPROVED** [review]
LGTM.

This will avoid spamming the debug log of SDK users.

My original jar hell fears are not accurate; those were for log4j1, so there is a distinction in classes.

We still have both api's as transitive dependencies but downstream users can exclude the one they don't want.

**Comment by dbwiddis** [comment]
> The change is failing the integ test though. You can remove the logger from IT test

That logger is log4j.  The problem is in the next line in the stack trace:
```
org.opensearch.sdk.sample.helloworld.rest.TestHelloWorldIT > testInitializeExtension FAILED
    java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
        at org.apache.hc.client5.http.ssl.AbstractClientTlsStrategy.<clinit>(AbstractClientTlsStrategy.java:64)
```
We are forced to import SLF4J logger here: https://hc.apache.org/httpcomponents-client-5.1.x/current/httpclient5/xref/org/apache/hc/client5/http/ssl/AbstractClientTlsStrategy.html#L64

We are forced to import LOG4J2 logger here:  https://github.com/opensearch-project/OpenSearch/blob/675acdb4b92d862ccec3ebd10f390f333d81291f/server/src/main/java/org/opensearch/common/settings/Settings.java#L857

We can therefore not exclude either API dependency from SDK without either:
 - removing any usage of the Settings class, or
 - removing any usage of Apache Client SSL classes

My original concern of forcing debug logging is resolved by moving the config file. 

Downstream users end up with two logging APIs but they can at least configure them and sort it out themselves.

**Review by owaiskazi19 — APPROVED** [review]
Fine with this temp fix!
